pkg/platform/provider/baremetal/validation/cluster.go

```diff
@@ -65,48 +66,95 @@ func ValidateClusterSpecVersion(version string, fldPath *field.Path, phase platf
 	return allErrs
 }
 
-// ValidateCIDR validates a given cidr.
-func ValidateCIDR(cidr string, fldPath *field.Path) field.ErrorList {
+func checkOverlapped(net1, net2 *net.IPNet) bool {
```

Comment:
Put this to util/net will be better. This file should only contains business logic.

Reply:
updated. Please check again.